### PR TITLE
#471 - Add ILM documentation

### LINKED ISSUE #471: Add ILM documentation
## Requirement - what kind of business use case are you trying to solve?

Add documentation relating to this PR: https://github.com/jaegertracing/jaeger/pull/2454

The changes should update the Elasticsearch deployment section, especially to Elasticsearch Rollover: https://www.jaegertracing.io/docs/1.21/deployment/#elasticsearch.

### PR DESCRIPTION
**Pull request by bhiravabhatla**
Signed-off-by: santosh <bsantosh@thoughtworks.com>

## Which problem is this PR solving?
- Resolves #471 

## Short description of the changes
- Adds documentation for enabling usage of ILM in elasticsearch for managing jaeger indices.

### CODE REVIEW

**Comment by bhiravabhatla** [comment]
@albertteoh, Kindly provide feedback.

File: content/docs/next-release/deployment.md
Comment (albertteoh) [inline content/docs/next-release/deployment.md:321]: I would aim to make the steps below as simple as possible for users to follow. A suggestion is to provide code that they can copy-paste where practical, even if it's just an example that they can modify later on, and avoid indirection. Having all the steps in one place just for ILM means we can avoid touching any existing manual rollover documentation.

I've provided a suggestion below.

File: content/docs/next-release/deployment.md
Comment (albertteoh) [inline content/docs/next-release/deployment.md:326]: I propose we replace this block with (and remove the changes from L336 onwards):

##### Enabling ILM support

1. Create an ILM policy in elasticsearch named *jaeger-ilm-policy*.

   For example, the following policy will rollover the "active" index when it is
   older than 1m and delete indices that are older than 2m.
   ```
   curl -X PUT http://localhost:9200/_ilm/policy/jaeger-ilm-policy \
   -H 'Content-Type: application/json; charset=utf-8' \
   --data-binary @- << EOF
   {
     "policy": {
       "phases": {
         "hot": {
           "min_age": "0ms",
           "actions": {
             "rollover": {
               "max_age": "1m"
             },
             "set_priority": {
               "priority": 100
             }
           }
         },
         "delete": {
           "min_age": "2m",
           "actions": {
             "delete": {}
           }
         }
       }
     }
   }
   EOF
   ```
2. Run elasticsearch initializer with `ES_USE_ILM=true`:
   ```
   docker run -it --rm --net=host -e ES_USE_ILM=true jaegertracing/jaeger-es-rollover:latest init http://localhost:9200 # <1>
   ```
   <1> If you need to initialize archive storage, add `-e ARCHIVE=true`.

   {{< info >}}
   While initializing with ILM support, make sure that an ILM policy named *jaeger-ilm-policy* is created in Elasticsearch beforehand.
   Initialization checks for this policy - if it does not exist, it would error out with following message.

   *ILM policy jaeger-ilm-policy doesn't exist in Elasticsearch. Please create it and rerun init*
   {{< /info >}}
3. After the initialization, deploy Jaeger with `--es.use-ilm=true` and `--es.use-aliases=true`.

File: content/docs/next-release/deployment.md
Comment (bhiravabhatla) [reply content/docs/next-release/deployment.md:309]: I had similar thought initially - but dint act on it. Agreed, will move it.

File: content/docs/next-release/deployment.md
Comment (bhiravabhatla) [reply content/docs/next-release/deployment.md:326]: Agreed, adding it. Just wanted mention something with respect to above ILM policy to be accurate, even though the max age in above policy is set to 1m,  rollover would only happen after 10m - as indices.lifecycle.poll_interval is set to 10 mins by default. 
This can be changed in cluster settings though. :)

**Comment by bhiravabhatla** [comment]
I also had a draft blog submitted for this in jaeger medium org. It was kept on hold until we had this PR merged. Can I update it with current approach and resubmit it for review, not sure if this is something you could help me with @albertteoh .

File: content/docs/next-release/deployment.md
Comment (albertteoh) [inline content/docs/next-release/deployment.md:400]:
```suggestion
  ```shell
```

File: content/docs/next-release/deployment.md
Comment (albertteoh) [inline content/docs/next-release/deployment.md:398]:
```suggestion
* Run elasticsearch initializer with `ES_USE_ILM=true`:
```

File: content/docs/next-release/deployment.md
Comment (albertteoh) [inline content/docs/next-release/deployment.md:410]:
```suggestion
  {{< info >}}
  While initializing with ILM support, make sure that an ILM policy named `jaeger-ilm-policy` is created in Elasticsearch beforehand (see the previous step), otherwise the following error message will be shown:

  "ILM policy jaeger-ilm-policy doesn't exist in Elasticsearch. Please create it and rerun init"
  {{< /info >}}
```

**Comment by albertteoh** [comment]
> I also had a draft blog submitted for this in jaeger medium org. It was kept on hold until we had this PR merged. Can I update it with current approach and resubmit it for review, not sure if this is something you could help me with @albertteoh .

I'm not sure if I have the privileges to review Jaeger medium posts (I haven't done one before), but happy to do so if it's possible for me.

File: content/docs/next-release/deployment.md
Comment (albertteoh) [inline content/docs/next-release/deployment.md:397]: unindent: line up with above `}`, otherwise the `EOF` will not work.

File: content/docs/next-release/deployment.md
Comment (albertteoh) [inline content/docs/next-release/deployment.md:402]: unindent by two spaces in line with above opening code block.

File: content/docs/next-release/deployment.md
Comment (albertteoh) [inline content/docs/next-release/deployment.md:411]:
```suggestion
  After the initialization, deploy Jaeger with `--es.use-ilm=true` and `--es.use-aliases=true`.
```

File: content/docs/next-release/deployment.md
Comment (albertteoh) [inline content/docs/next-release/deployment.md:373]:
```suggestion
  curl -X PUT http://localhost:9200/_ilm/policy/jaeger-ilm-policy \
  -H 'Content-Type: application/json; charset=utf-8' \
  --data-binary @- << EOF
  {
```

File: content/docs/next-release/deployment.md
Comment (albertteoh) [inline content/docs/next-release/deployment.md:367]:
```suggestion
  For example, the following policy will rollover the "active" index when it is
  older than 1m and delete indices that are older than 2m.
```

File: content/docs/next-release/deployment.md
Comment (bhiravabhatla) [reply content/docs/next-release/deployment.md:397]: Yep, my bad. Done and checked the curl myself.

**Review by albertteoh — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**
